### DISCUSSION TITLE
Plugin Support/Implementation

### DISCUSSION THREAD

**Discussion by G4PLS** [streamdeck-linux-gui/streamdeck-linux-gui#142] [Ideas / Feature Requests]
In here I just want to talk about the Scope of Plugins and how someone could implement them. All of this is based on the message in [this](https://github.com/streamdeck-linux-gui/streamdeck-linux-gui/pull/133) pull request
I have the basics but its currently not that good and I want to try and make an actual Plugin implementation that is easy to use, or at least as easy to use as possible.

**There are a few points that came to mind and were mentioned in the pull request like:**
- How should UI be handled
- How to get them Running in the actual Program
- Overall Scope of Plugins

**Comment by coolapso** [comment]
Some talks have been taken in the background about this topic and its very much something we would like to have in the application.  However according to those who are more familiar with the code base, in its current state its a huge endeavor and the effort to make it work is quite hard. 

Some options on the table: 

* Decouple the UI from the core application, and re-write the UI in some other way (electron, web-ui) 
* Re-write the whole thing!! Yes you heard it right! Re-write the whole thing from the ground up (libraries included) in a language more familiar to the core maintainers, with easier dependency management and easier to install for the users.
